### PR TITLE
Ensure you can run Pulp without /var/lib/pulp

### PR DESCRIPTION
**Pull request by pavelpicka**
User can change where files are stored which help
to use with openshift and others cloud systems.

closes: #4380
https://pulp.plan.io/issues/4380

Signed-off-by: Pavel Picka <ppicka@redhat.com>

### CODE REVIEW

**Comment by pavelpicka** [comment]
Testing of this pull request requires changes of MEDIA_ROOT in '/etc/pulp/settings.py' (with pulp offline) and remounting or moving media directory out of '/var/lib/pulp'.

File: pulpcore/settings_helper.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+            working_path = os.path.join(media_path, 'tmp/')
+    return {'MEDIA_ROOT': media_path,
+            'STATIC_ROOT': static_path,
+            'WORKING_DIRECTORY': working_path}
+
+
+def get_settings_value(value):
+    """Returns requested path"""
+    if value in prepare_paths():
+        return prepare_paths()[value]
```
Comment (daviddavis) [inline pulpcore/settings_helper.py]: I'm not sure I understand the purpose of `prepare_paths()` and `get_settings_value(value)`?

File: pulpcore/app/settings.py [hunk 2]
```diff
@@ -12,11 +12,14 @@
...
 from pkg_resources import iter_entry_points
+from pulpcore import settings_helper
 
 
 # Build paths inside the project like this: os.path.join(BASE_DIR, ...)
 BASE_DIR = os.path.dirname(os.path.abspath(__file__))
-STATIC_ROOT = '/var/lib/pulp/static/'
+MEDIA_ROOT = settings_helper.get_settings_value('MEDIA_ROOT')
+STATIC_ROOT = settings_helper.get_settings_value('STATIC_ROOT')
+WORKING_DIRECTORY = settings_helper.get_settings_value('WORKING_DIRECTORY')
```
Comment (daviddavis) [inline pulpcore/app/settings.py]: Why not just define these as:

```python
MEDIA_ROOT = '/var/lib/pulp/'
STATIC_ROOT = os.path.join(MEDIA_ROOT, 'static/')
WORKING_DIRECTORY = os.path.join(media_path, 'tmp')
```

Comment (pavelpicka) [reply pulpcore/app/settings.py]: In settings.py there is not yet loaded config via dynaconf so to let user change media_root path I used custom loader. 

get_settings_value function is for when user specify all 3 path differently (or just one which is not media_root), like 
```
MEDIA_ROOT = /pulp3
STATIC_ROOT = /static_files/pulp3
WORKING_DIRECTORY = /tmp
```

**Comment by pavelpicka** [comment]
Rebased on @dkliban #35 and tested with pulp_file plugin

**Pull request merged**